aplib.net-demo/Assets/Scripts/MainMenu/MenuButtons.cs

```diff
@@ -1,48 +1,113 @@
 using UnityEngine;
 using UnityEngine.SceneManagement;
 
-public class MenuPlayButton : MonoBehaviour
+/// <summary>
+/// This class houses all the settings and menu UI buttons/methods.
+/// It is called CanvasManager as it manages all the UI canvi.
+/// </summary>
+public class CanvasManager : MonoBehaviour
 {
     /// <summary>
     /// Reference to the menu canvas.
     /// </summary>
     public GameObject menuCanvas;
+    /// <summary>
+    /// Reference to the settings canvas of the menu.
+    /// </summary>
+    public GameObject settingMenuCanvas;
+    /// <summary>
+    /// Reference to the settings canvas of the game.
+    /// </summary>
+    public GameObject settingGameCanvas;
 
     /// <summary>
-    /// Reference to the settings canvas.
+    /// To ensure the menu settings and menu UI aren't on on the same time.
+    /// </summary>
+    public bool isOnMenuSettings = false;
+    /// <summary>
+    /// To ensure the game settings and menu UI aren't on on the same time.
     /// </summary>
-    public GameObject settingCanvas;
+    public bool isOnGameSettings = false;
 
     /// <summary>
-    /// To ensure the settings and menu UI aren't on on the same time.
+    /// This bool communicates with the mouse lock script to enable and disable the curser.
+    /// </summary>
+    public bool IsCursurNeeded => isOnGameSettings;
+
+    /// <summary>
+    /// This string keeps track of what scene we are in.
+    /// </summary>
+    string _currentSceneName = "";
```

Comment:
please add private to fields without access modifiers
```suggestion
    private string _currentSceneName = "";
```